File: julia/det.jl
# -*- coding: utf-8 -*-

using Einsum

eijk = zeros(3, 3, 3)
eijk[1, 2, 3] = eijk[2, 3, 1] = eijk[3, 1, 2] = 1
eijk[1, 3, 2] = eijk[3, 2, 1] = eijk[2, 1, 3] = -1

# A = transpose(reshape(1:9, (3, 3)))
A = randn(3, 3)
det_julia = det(A)
A1 = A[:, 1]
A2 = A[:, 2]
A3 = A[:, 3]
@einsum det_einsum := eijk[i,j,k] * A1[i] * A2[j] * A3[k]
println(det_julia)
println(det_einsum)    

File: python/det.py
# -*- coding: utf-8 -*-

import numpy as np

eijk = np.zeros((3, 3, 3))
eijk[0, 1, 2] = eijk[1, 2, 0] = eijk[2, 0, 1] = 1
eijk[0, 2, 1] = eijk[1, 0, 2] = eijk[2, 1, 0] = -1


A = np.arange(9).reshape(3, 3)
det_numpy = np.linalg.det(A)
det_einsum = np.einsum("ijk,i,j,k", eijk, A[0], A[1], A[2])
print(det_numpy, det_einsum)
